File: Output/c/medium/SumOfSubarrayMinimums.c
/**https://leetcode.com/problems/sum-of-subarray-minimums/ */
//Given an array of integers arr, find the sum of min(b), where b ranges over every (contiguous) subarray of arr. Since the answer may be large, return the answer modulo 109 + 7.


int sumSubarrayMins(int* arr, int arrSize){
    int result = 0;
    int dp[arrSize][arrSize];
    for(int i = 0; i < arrSize; i++){
        for(int j = 0; j < arrSize; j++){
            dp[i][j] = 0;
        }
    }
    for(int i = 0; i < arrSize; i++){
        dp[i][i] = arr[i];
        result = result + arr[i];
    }
    for(int i = 1; i < arrSize; i++){
        for(int j = 0; j < arrSize - i; j++){
            dp[j][j + i] = dp[j][j + i - 1] < arr[j + i] ? dp[j][j + i - 1] : arr[j + i];
            result = result + dp[j][j + i];
        }
    }
    return result;

}

    

File: Output/python/medium/SumOfSubarrayMinimums.py
#https://leetcode.com/problems/sum-of-subarray-minimums/
#Given an array of integers arr, find the sum of min(b), where b ranges over every (contiguous) subarray of arr. Since the answer may be large, return the answer modulo 109 + 7.


class Solution(object):
    def sumSubarrayMins(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """       
        res = 0
        for i in range(len(arr)):
            for j in range(i, len(arr)):
                res += min(arr[i:j+1])
        return res % (10**9 + 7)

        

    